etsin_finder/frontend/js/components/general/modal.jsx

```diff
@@ -71,41 +71,54 @@ export default class Modal extends Component {
     this.state = {
       styles: { overlay: overlayStyle, content: contentStyle },
     }
+  }
+
+  componentDidMount() {
+    this.updateBlur()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.isOpen !== prevProps.isOpen) {
+      this.updateBlur()
+    }
+  }
 
-    this.afterOpen = this.afterOpen.bind(this)
-    this.onClose = this.onClose.bind(this)
+  componentWillUnmount() {
+    this.hideBlur()
   }
 
-  onClose() {
+  showBlur() {
     const root = document.getElementById('root')
-    root.classList.remove('blur')
-    this.props.onRequestClose()
+    if (root) {
+      root.classList.add('blur')
+    }
   }
 
-   // eslint-disable-next-line camelcase
-   UNSAFE_componentWillReceiveProps(newProps) {
```

Comment:
Very good that you refactored the code to not use the componentWillReceiveProps and handle changes with componentWillMount and UnMount, nice.

Reply:
Nice. Since I'm using the <Modal> component now too, I heard from @kasperkeinanen about this improvement. Very good!